### PR TITLE
[BEAM-2632] Use Junit Paramaterized test suits in TextIOReadTest

### PR DESCRIPTION
**Pull request by huafengw**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by manuzhang** [comment]
retest this please

**Comment by huafengw** [comment]
R: @kennknowles

**Comment by manuzhang** [comment]
R: @jkff Please take a look.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOReadTest.java [hunk 1]
```diff
@@ -462,547 +217,660 @@ private String createZipFile(List<String> expected, String filename, String[]...
...
+    }
 
-  @Test
-  public void testSplittingSourceWithCarriageReturnDelimiterAndNonEmptyBytesAtEnd()
-      throws Exception {
-    TextSource source = prepareSource("asdf\rhjkl\rxyz".getBytes(UTF_8));
-    SourceTestUtils.assertSplitAtFractionExhaustive(source, PipelineOptionsFactory.create());
-  }
+    /**
+     * Tests reading from a small, uncompressed file with .gz extension. This must work in AUTO or
```
Comment (manuzhang) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOReadTest.java]: Why "or" ? It seems clear that this is to test GZIP modes and the one below AUTO modes

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOReadTest.java [hunk 2]
```diff
@@ -462,547 +217,660 @@ private String createZipFile(List<String> expected, String filename, String[]...
...
-    SourceTestUtils.assertSplitAtFractionExhaustive(source, PipelineOptionsFactory.create());
-  }
+    /**
+     * Tests a zip file with no entries. This is a corner case not tested elsewhere as the default
+     * test zip files have a single entry.
+     */
+    @Test
+    @Category(NeedsRunner.class)
+    public void testZipCompressedReadWithNoEntries() throws Exception {
+      File filename = createZipFile(new ArrayList<String>(), tempFolder, "empty zip file");
```
Comment (manuzhang) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOReadTest.java]: better name it "file"

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOReadTest.java [hunk 3]
```diff
@@ -462,547 +217,660 @@ private String createZipFile(List<String> expected, String filename, String[]...
...
+    @Test
+    @Category(NeedsRunner.class)
+    public void testZipCompressedReadWithMultiEntriesFile() throws Exception {
+      String[] entry0 = new String[] {"first", "second", "three"};
+      String[] entry1 = new String[] {"four", "five", "six"};
+      String[] entry2 = new String[] {"seven", "eight", "nine"};
+
+      List<String> expected = new ArrayList<>();
+
+      File filename =
```
Comment (manuzhang) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOReadTest.java]: ditto

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOReadTest.java [hunk 4]
```diff
@@ -462,547 +217,660 @@ private String createZipFile(List<String> expected, String filename, String[]...
...
-        EmptyMatchTreatment.DISALLOW, delimiter);
-  }
+    /**
+     * Read a ZIP compressed file containing data, multiple empty entries, and then more data. We
+     * expect just the data back.
+     */
+    @Test
+    @Category(NeedsRunner.class)
+    public void testZipCompressedReadWithComplexEmptyAndPresentEntries() throws Exception {
+      File filename =
```
Comment (manuzhang) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOReadTest.java]: ditto

**Comment by manuzhang** [comment]
retest this please

**Pull request closed without merging**